rust/src/dns/dns.rs

```diff
@@ -322,7 +322,8 @@ pub struct DNSState {
 
 impl State<DNSTransaction> for DNSState {
     fn get_transactions(&self) -> &[DNSTransaction] {
-        &self.transactions
+        let (r, _) = self.transactions.as_slices();
+        return r;
```

Comment:
This would only return a portion of the transactions. You probably have to use `make_contiguous`.  Can you test that with your fuzzing?

Reply:
Strange that it did not fail CI if it was really needed...

Reply:
By the way, `make_contiguous` needs to make `self` mutable...

Reply:
> Strange that it did not fail CI if it was really needed...

If all removes are from the front, and given all adds are to the back, in most cases I think we'd be in a state where we might only use one half?

Still worth figuring out if we're to use a VecDeque, as mentioned I see it having a negligible affect in normal conditions, however the fuzz appears to produce a worst case which would be nice to fixup.